### DISCUSSION TITLE
Storing data in custom struct and bind it to view

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [slint-ui/slint#3959] [Q&A] [answered]
How would I actually store data from a custom struct and then bind it to the view?

I'm so used to XAML languages with bindings, models and viewmodels (MVVM).
How different is it using slint? I am using the Rust language. Can I just create a custom struct in Rust and then import said datatype or do I have to use the slint declarative macro (`slint::slint! { .. }`) ?

**Comment by ogoffart** [comment, marked as answer]
Not sure exactly what you have in mind. 
The Slint compiler needs to know about all the types,  so you should create `struct` in the slint! macro or in .slint files.
